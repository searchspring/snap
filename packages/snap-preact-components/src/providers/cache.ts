export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
