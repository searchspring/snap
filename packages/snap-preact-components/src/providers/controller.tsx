import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
