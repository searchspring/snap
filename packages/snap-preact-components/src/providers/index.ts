export * from './store';
export * from './theme';
