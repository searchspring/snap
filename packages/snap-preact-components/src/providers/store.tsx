import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractStore } from '@searchspring/snap-store-mobx';

const StoreContext = createContext<null | AbstractStore>(null);

export const StoreProvider = ({ children, store }: { children: ComponentChildren; store: AbstractStore }) => {
	return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};

export const useStore = () => useContext(StoreContext);

export function withStore<C extends ComponentType>(Component: C): C {
	return ((props: any) => <StoreContext.Consumer>{(store) => <Component {...props} store={store} />}</StoreContext.Consumer>) as C;
}
