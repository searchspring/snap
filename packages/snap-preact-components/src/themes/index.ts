export * from './pike';
export * from './bocachica';
