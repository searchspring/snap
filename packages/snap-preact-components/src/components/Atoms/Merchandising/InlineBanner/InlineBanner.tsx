/** @jsx jsx */
import { Fragment, h } from 'preact';

import { jsx, css } from '@emotion/react';
import classnames from 'classnames';
import { Theme, useTheme, CacheProvider } from '../../../../providers';
import { ComponentProps, ResultsLayout, ResultsLayoutType, StylingCSS } from '../../../../types';
import { mergeProps } from '../../../../utilities';

import type { Banner } from '@searchspring/snap-store-mobx';

const CSS = {
	inlineBanner: ({ width }: Partial<InlineBannerProps>) =>
		css({
			height: '100%',
			display: 'flex',
			flexDirection: 'column',
			justifyContent: 'center',
			alignItems: 'center',
			width: width,
			'&.ss__inline-banner--grid': {
				flexDirection: 'column',
			},
			'&.ss__inline-banner--list': {
				flexDirection: 'row',
				display: 'block',
				width: '100%',
			},
			'& iframe': {
				maxWidth: '100%',
			},
		}),
};

export function InlineBanner(properties: InlineBannerProps): JSX.Element {
	const globalTheme: Theme = useTheme();
	const theme = { ...globalTheme, ...properties.theme };
	const defaultProps: Partial<InlineBannerProps> = {
		layout: ResultsLayout.GRID,
		width: 'auto',
	};

	const props = mergeProps('inlineBanner', globalTheme, defaultProps, properties);

	const { banner, disableStyles, className, layout, onClick, style, styleScript } = props;

	const styling: { css?: StylingCSS } = {};
	const stylingProps = { ...props, theme };

	if (styleScript) {
		styling.css = [styleScript(stylingProps), style];
	} else if (!disableStyles) {
		styling.css = [CSS.inlineBanner(stylingProps), style];
	} else if (style) {
		styling.css = [style];
	}

	return banner && banner.value ? (
		<CacheProvider>
			<div
				onClick={(e: React.MouseEvent<Element, MouseEvent>) => {
					onClick && onClick(e, banner);
				}}
				className={classnames('ss__inline-banner', `ss__inline-banner--${layout}`, className)}
				{...styling}
				dangerouslySetInnerHTML={{
					__html: banner.value,
				}}
			/>
		</CacheProvider>
	) : (
		<Fragment></Fragment>
	);
}

export interface InlineBannerProps extends ComponentProps {
	banner: Banner;
	width?: string;
	layout?: ResultsLayoutType;
	onClick?: (e: React.MouseEvent, banner: Banner) => void;
}
