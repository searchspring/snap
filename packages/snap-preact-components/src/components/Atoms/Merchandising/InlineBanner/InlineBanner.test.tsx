import { h } from 'preact';

import { render } from '@testing-library/preact';
import userEvent from '@testing-library/user-event';

import { InlineBanner } from './InlineBanner';
import { ThemeProvider } from '../../../../providers';
import themes from '../../../../themes';
import type { Banner } from '@searchspring/snap-store-mobx';

import { MockData } from '@searchspring/snap-shared';
import { SearchResponseModel } from '@searchspring/snapi-types';

const mockData = new MockData();
mockData.updateConfig({ search: 'inlineBanners' });
const searchResponse: SearchResponseModel = mockData.search();

describe('Merchandising Inline Banner Component', () => {
	Object.keys(themes || {}).forEach((themeName) => {
		it(`uses ${themeName} theme`, () => {
			const theme = themes[themeName as keyof typeof themes];
			const rendered = render(<InlineBanner theme={theme} banner={searchResponse.merchandising?.content?.inline![0] as Banner} />);
			expect(rendered.asFragment()).toMatchSnapshot();
		});
	});

	it('renders type:inline banner', () => {
		const rendered = render(<InlineBanner banner={searchResponse.merchandising?.content?.inline![0] as Banner} />);
		const merchBannerElement = rendered.container.querySelector('.ss__inline-banner');
		expect(merchBannerElement).toBeInTheDocument();
		expect(merchBannerElement?.innerHTML).toBe(searchResponse.merchandising?.content?.inline![0].value!);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can set a custom onClick function', () => {
		const onClickFunc = jest.fn();

		const rendered = render(<InlineBanner banner={searchResponse.merchandising?.content?.inline![0] as Banner} onClick={onClickFunc} />);
		const merchBannerElement = rendered.container.querySelector('.ss__inline-banner')!;
		expect(merchBannerElement).toBeInTheDocument();
		expect(rendered.asFragment()).toMatchSnapshot();

		userEvent.click(merchBannerElement);
		expect(onClickFunc).toHaveBeenCalled();
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('can disable styling', () => {
		const rendered = render(<InlineBanner disableStyles={true} banner={searchResponse.merchandising?.content?.inline![0] as Banner} />);
		const loadingbarElement = rendered.container.querySelector('.ss__inline-banner');
		expect(loadingbarElement?.classList.length).toBe(2);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('renders with classname', () => {
		const className = 'classy';
		const rendered = render(<InlineBanner className={className} banner={searchResponse.merchandising?.content?.inline![0] as Banner} />);
		const merchBannerElement = rendered.container.querySelector('.ss__inline-banner');
		expect(merchBannerElement).toBeInTheDocument();
		expect(merchBannerElement).toHaveClass(className);
		expect(rendered.asFragment()).toMatchSnapshot();
	});
});

describe('InlineBanner theming works', () => {
	it('is themeable with ThemeProvider', () => {
		const globalTheme = {
			components: {
				inlineBanner: {
					className: 'theme-class',
				},
			},
		};

		const rendered = render(
			<ThemeProvider theme={globalTheme}>
				<InlineBanner banner={searchResponse.merchandising?.content?.inline![0] as Banner} />
			</ThemeProvider>
		);

		const InlineBannerElement = rendered.container.querySelector('.ss__inline-banner');
		expect(InlineBannerElement).toHaveClass(globalTheme.components.inlineBanner.className);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable with theme prop', () => {
		const propTheme = {
			components: {
				inlineBanner: {
					className: 'theme-class',
				},
			},
		};

		const rendered = render(<InlineBanner banner={searchResponse.merchandising?.content?.inline![0] as Banner} theme={propTheme} />);

		const InlineBannerElement = rendered.container.querySelector('.ss__inline-banner');
		expect(InlineBannerElement).toHaveClass(propTheme.components.inlineBanner.className);
		expect(rendered.asFragment()).toMatchSnapshot();
	});

	it('is themeable and theme prop overrides ThemeProvider', () => {
		const globalTheme = {
			components: {
				inlineBanner: {
					className: 'global-class',
				},
			},
		};

		const propTheme = {
			components: {
				inlineBanner: {
					className: 'theme-class',
				},
			},
		};

		const rendered = render(
			<ThemeProvider theme={globalTheme}>
				<InlineBanner banner={searchResponse.merchandising?.content?.inline![0] as Banner} theme={propTheme} />
			</ThemeProvider>
		);

		const InlineBannerElement = rendered.container.querySelector('.ss__inline-banner');
		expect(InlineBannerElement).toHaveClass(propTheme.components.inlineBanner.className);
		expect(InlineBannerElement).not.toHaveClass(globalTheme.components.inlineBanner.className);
		expect(rendered.asFragment()).toMatchSnapshot();
	});
});
