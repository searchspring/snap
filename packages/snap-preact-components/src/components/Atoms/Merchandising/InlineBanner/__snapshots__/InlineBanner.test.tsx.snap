// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineBanner theming works is themeable and theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid theme-class emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InlineBanner theming works is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid theme-class emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InlineBanner theming works is themeable with theme prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid theme-class emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Merchandising Inline Banner Component can disable styling 1`] = `
<DocumentFragment>
  <div
    class="ss__inline-banner ss__inline-banner--grid"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-0\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-0"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Merchandising Inline Banner Component can set a custom onClick function 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Merchandising Inline Banner Component can set a custom onClick function 2`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Merchandising Inline Banner Component renders type:inline banner 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Merchandising Inline Banner Component renders with classname 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid classy emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Merchandising Inline Banner Component uses bocachica theme 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
}

.emotion-0.ss__inline-banner--grid {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0.ss__inline-banner--list {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: block;
  width: 100%;
}

.emotion-0 iframe {
  max-width: 100%;
}

<div
    class="ss__inline-banner ss__inline-banner--grid emotion-0"
  >
    <script
      data-banner-html="<div	style=\\"		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	\\">	<div		style=\\"			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		\\"	>		<div style=\\"padding: 20px\\">			<h3				class=\\"emotion-1\\"				style=\\"					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				\\"			>				It's okay to be cool in Winter.			</h3>			<p style=\\"color: #fff; font-size: 16px\\">				<strong>20% off select sunglasses.</strong> <br />Limited time offer.			</p>		</div>	</div></div>"
      data-banner-type="html"
      type="text/widget"
    />
    <div
      style="		background-image: url('http://searchspring-demo-content.s3.amazonaws.com/demo/fashion/product_images_large/478a0515_3_large.jpg');		background-position: center;		background-size: cover;		height: 100%;		width: 100%;	"
    >
      	
      <div
        style="			background: rgba(0, 0, 0, 0.4);			display: flex;			flex-direction: column;			justify-content: flex-end;			height: 100%;		"
      >
        		
        <div
          style="padding: 20px"
        >
          			
          <h3
            class="emotion-1"
            style="					font-size: 36px;					font-weight: 100;					color: #fff;					line-height: 1.2em;				"
          >
            				It's okay to be cool in Winter.			
          </h3>
          			
          <p
            style="color: #fff; font-size: 16px"
          >
            				
            <strong>
              20% off select sunglasses.
            </strong>
             
            <br />
            Limited time offer.			
          </p>
          		
        </div>
        	
      </div>
    </div>
  </div>
</DocumentFragment>
`;
